knimin/lib/data_access.py

```diff
@@ -3110,6 +3134,59 @@ def write_sample_plate_layout(self, sample_plate_id, layout):
             TRN.add(sql, sql_args, many=True)
             TRN.execute()
 
+    def get_blanks_from_sample_plate(self, sample_plate_id):
+        """Returns the blanks from the sample plate
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            ID of the sample plate whose layout is to be read
+
+        Returns
+        -------
+        list of DictCursor
+            The sample id, the row and the col
+        """
+        with TRN:
+            self._sample_plate_exists(sample_plate_id)
+
+            sql = """SELECT sample_id, row, col
+                     FROM pm.sample_plate_layout
+                        JOIN pm.sample USING (sample_id)
+                     WHERE sample_plate_id = %s AND is_blank=true
+                     ORDER BY row, col"""
+            TRN.add(sql, [sample_plate_id])
+            return TRN.execute_fetchindex()
+
+    def get_replicates_from_sample_plate(self, sample_plate_id):
+        """Returns the technical replicates present in the sample plate
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            Sample plate id
+
+        Returns
+        -------
+        dict
+            For each sample, a list of tuples with the well locations
+        """
+        with TRN:
+            self._sample_plate_exists(sample_plate_id)
+            sql = """SELECT sample_id, ARRAY_AGG((row, col)) AS wells
+                     FROM pm.sample_plate_layout
+                        JOIN pm.sample USING (sample_id)
+                     WHERE sample_plate_id = %s
+                        AND is_blank = false
+                     GROUP BY sample_id"""
+            TRN.add(sql, [sample_plate_id])
+            replicates = {}
+            for sql_row in TRN.execute_fetchindex():
+                wells = eval(sql_row['wells'])
+                if len(wells) > 1:
+                    replicates[sql_row['sample_id']] = [eval(i) for i in wells]
```

Comment:
Unfortunately it is not, the first eval only returns a list of strings, so I need the second one to actually break up the string into the tuple. I think is due to the format that SQL is sending the data back